proposals/001_interop.md

```diff
@@ -0,0 +1,473 @@
+`@context` key for users to interpret as a JSON-LD object. MUST NOT return a top-level
+`@graph` key.
+
+```json
+ {
+   "@context": "https://w3id.org/docmaps/context.jsonld",
+   "name": "John Doe",
+   "id": "https://example.com/ex/docmaps/v1/publisher/example_press",
+   "homepage": "https://example.com/",
+   "logo": "https://example.com/logo.img",
+   "account": {
+     "id": "https://github.com/docmaps-project",
+     "service": "https://github.com"
+   }
+ },
+```
+
+where
+
+PATH | DETAILS
+| - | - |
+| `@context` |  If present, MUST be a JSON-LD context. SHOULD be exactly `"https://w3id.org/docmaps/context.jsonld"`  |
+| `id` |  The ID of the publisher. MUST be an IRI and MUST be a URL which when queried responds with this publisher exactly. |
+
+Publishers are always named nodes (nodes with IRIs for named nodes (nodes with IRIs) in the Docmaps specification. Therefore
+they are always considered persistent. Any signatory information is separate from  their RDF bodies (the data in this  endpoint's
+response), so those bodies can safely  change. However, if the IRI of a publisher changes, clients will need to handle the migration
+by their own logic until  the core Docmaps protocol is improved ([see Github  Issue here](https://github.com/Docmaps-Project/docmaps/issues/69)).
+That  improvement is out of the scope of this RFC.
+
+The authors of this specification anticipate the relationship between a server and specific Publishers whose signing identities
+are owned by the server may be expressed in extensions to this endpoint. See "Security" below.
+
+### Searching for docmaps based on content matter
+
+These endpoints are REQUIRED for all standard Docmaps servers. They enable consumers to discover what docmaps or objects
+can be retrieved from this server. They return information about further quueries that can be performed to retrieve docmaps.
+
+#### POST /search
+
+The search query MUST be accepted in the request body as type `application/json`. Although servers are invited to
+decide on search semantics that are most appropriate to their use case, servers MUST support the following body
+fields in the search:
+
+```json
+{
+  "query_terms": [
+    {
+      "match": "https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/index",
+      "paths": ["publisher.id"]
+    },
+    {
+      "match": "docmap",
+      "paths": ["type"]
+    }
+  ]
+}
+```
+
+where
+
+PATH | DETAILS
+| - | - |
+| `query_terms` | MUST be an array of object. |
+| `query_terms[].match` | MUST be a full IRI or a JSON-LD shortcut present in the Docmaps JSON-LD context. |
+| `query_terms[].paths` | MUST be an array of JSON paths, IRIs or JSON-LD shortcuts that the match term matches to. |
+
+**Response** MUST return a linked-data (JSON-LD) graph, because this endpoint is not strictly typed.
+For use-cases where JSON-LD is of limited applicability, the graph may be disjoint (i.e.,
```

Comment:
Can the response also include the full DocMaps?

Reply:
Yes, the response to these queries can be Docmaps, but since they can also return partial docmaps, there is a normalized response form where the top level key is `@graph`. It can be a graph with one object which is a framed Docmap, and parties might privately agree to assume this, but clients should use JSON-LD processors to read the response when possible.